streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java

```diff
@@ -197,6 +198,49 @@ public void shouldNotRegisterStoreWithoutMetadata() {
             () -> changelogReader.register(new TopicPartition("ChangelogWithoutStoreMetadata", 0), stateManager));
     }
 
+    @Test
+    public void shouldSupportUnregisterChangelogBeforeCompletion() {
```

Comment:
We should only call this is we are in `RESTORING` state, right? If we are `COMPLETED`, we should not call `suspend` because we did call `end` already? So the test name seems missleading?

Reply:
Should we add a case for a "completed" partition, and that we don't call `suspended` during `unregister` for this case?

Reply:
I can add that.